src/commands/transaction/accountoperationrestriction.ts

```diff
@@ -0,0 +1,165 @@
+    })
+    restrictionType: string;
+
+    @option({
+        flag: 'd',
+        description: '(Optional) incoming/outgoing (blank means incoming)',
+    })
+    restrictionDirection: string;
+
+    @option({
+        flag: 'a',
+        description: 'Modification action. (1: Add, 0: Remove)',
+        validator: new ModificationActionValidator(),
+    })
+    modificationAction: string;
+
+    @option({
+        flag: 'v',
+        description: 'Transaction Type [' + CommandOptions.transactionType + ']',
+    })
+    value: string;
+
+    @option({
+        flag: 'f',
+        description: '(Optional) Maximum fee',
+    })
+    maxfee: number;
+}
+
+@command({
+    description: 'Allow or block outgoing transactions by transaction type.',
+})
+export default class extends ProfileCommand {
+    constructor() {
+        super();
+    }
+
+    @metadata
+    execute(options: CommandOptions) {
+        options.restrictionType = OptionsResolver(options,
+            'restrictionType',
+            () => undefined,
+            'Fill in the restriction type (allow / block): ');
+
+        options.modificationAction = OptionsResolver(options,
+            'modificationAction',
+            () => undefined,
+            'Fill in the modification action (1: Add, 0: Remove): ');
+
+        if (!CommandOptions.transactionType.includes(options.value)) {
+            options.value = OptionsResolver(options,
+                'value',
+                () => undefined,
+                'Fill in the transaction Type (' + CommandOptions.transactionType + '): ');
+        }
+
+        let modificationAction;
+        if ('1' === options.modificationAction) {
+            modificationAction = AccountRestrictionModificationAction.Add;
+        } else if ('0' === options.modificationAction) {
+            modificationAction = AccountRestrictionModificationAction.Remove;
+        } else {
+            console.log('Wrong modificationAction. ModificationAction must be one of 1 or 0');
+            return;
+        }
+
+        let restrictionType;
+        if ('allow' === options.restrictionType.toLowerCase()) {
+            if ('outgoing' === options.restrictionDirection.toLowerCase()) {
+                restrictionType = AccountRestrictionType.AllowOutgoingTransactionType;
+            } else {
+                restrictionType = AccountRestrictionType.AllowIncomingTransactionType;
+            }
+        } else if ('block' === options.restrictionType.toLowerCase()) {
+            if ('outgoing' === options.restrictionDirection.toLowerCase()) {
+                restrictionType = AccountRestrictionType.BlockOutgoingTransactionType;
+            } else {
+                restrictionType = AccountRestrictionType.BlockIncomingTransactionType;
+            }
+        } else {
+            console.log('Wrong restrictionType. restrictionType must be one of \'allow\' or \'block\'');
+            return;
+        }
+
+        let transactionEntity;
```

Comment:
TransactionType['KEY'] should allow us to remove this switch

Reply:
variable `TransactionType` is an Enum type,I can't use ['key'] format